### ISSUE TITLE
BUG: MultiIndex.from_arrays doesn't raise if arrays are unequal lengths

### ISSUE THREAD

**Issue by TomAugspurger** [pandas-dev/pandas#10127]
``` python
In [62]: index = pd.MultiIndex.from_arrays([['a', 'b', 'c'], [1, 2]])

In [63]: index
Out[63]:
MultiIndex(levels=[['a', 'b', 'c'], [1, 2]],
           labels=[[0, 1, 2], [0, 1]])

```

Probably not what the user wants, and we rely on them being the same length in, e.g. `index.shape` which will raise a ValueError if you access it.

**Comment by mroeschke** [comment]
This looks like a duplicate of #13599 which is closed by #13728.

**Comment by TomAugspurger** [comment]
Indeed. Thanks!
